lib/gssapi/mech/gss_aeap.c

```diff
@@ -214,3 +214,121 @@ gss_context_query_attributes(OM_uint32 *minor_status,
 
     return GSS_S_FAILURE;
 }
+
+/*
+ * AEAD wrap API for a single piece of associated data, for compatibility
+ * with MIT and as specified by draft-howard-gssapi-aead-00.txt.
+ *
+ * @ingroup gssapi
+ */
+GSSAPI_LIB_FUNCTION OM_uint32 GSSAPI_LIB_CALL
+gss_wrap_aead(OM_uint32 *minor_status,
+	      gss_ctx_id_t context_handle,
+              int conf_req_flag,
+              gss_qop_t qop_req,
+              gss_buffer_t input_assoc_buffer,
```

Comment:
Sure, makes sense. Actually porting mech_eap to current Heimdal was a bit of a pain owing to all the const changes - I wasn't entirely clear if const context and cred handles made sense always, eg sequence numbers changing in contexts.

Sent from my iPhone

> On 6 Feb 2016, at 10:35, Nico Williams <notifications@github.com> wrote:
> 
> In lib/gssapi/mech/gss_aeap.c:
> 
> > @@ -214,3 +214,121 @@ gss_context_query_attributes(OM_uint32 *minor_status,
> >  
> >      return GSS_S_FAILURE;
> >  }
> > +
> > +/*
> > + * AEAD wrap API for a single piece of associated data, for compatibility
> > + * with MIT and as specified by draft-howard-gssapi-aead-00.txt.
> > + *
> > + * @ingroup gssapi
> > + */
> > +GSSAPI_LIB_FUNCTION OM_uint32 GSSAPI_LIB_CALL
> > +gss_wrap_aead(OM_uint32 *minor_status,
> > +	      gss_ctx_id_t context_handle,
> > +              int conf_req_flag,
> > +              gss_qop_t qop_req,
> > +              gss_buffer_t input_assoc_buffer,
> Should the inputs be gss_const_buffer_t? They aren't in MIT krb5, but I think they should be.
> 
> —
> Reply to this email directly or view it on GitHub.
>